# Tipos de datos

https://github.com/igijon/python_bases/tree/main/1_Tipos_Datos

## Variables

In [1]:
nombre = "Juan"
print(nombre)
nombre = "Laura"
print(nombre)

edad = 13
print(nombre, edad)

nombre = input("Dime tu nombre")
print("Tu nombre es "+nombre)

nombre1 = "Hola "
nombre2 = "Python_Bases"
frase = nombre1+nombre2

print(frase)

edad = 30
edad2 = 40
total = edad+edad2
print(total)

Juan
Laura
Laura 13
Tu nombre es Inma
Hola Python_Bases
70


## Formatear cadenas

In [2]:
x = 10
y = 5

print("Mis números son "+str(x)+" y "+str(y))
print("Mis números son {} y {}".format(x, y))
print("La suma de {} y {} es igual a {}".format(x, y, x+y))
print(f"Mis números son {x} y {y}")  # Cadenas literales

color = "rojo"
matricula = 54345
print(f"El coche es de color {color} y su matrícula es {matricula}")  # Cadenas literales

Mis números son 10 y 5
Mis números son 10 y 5
La suma de 10 y 5 es igual a 15
Mis números son 10 y 5
El coche es de color rojo y su matrícula es 54345


## Operadores

In [3]:
x = 6
y = 2
z = 7
print(f"{x} más {y} es {x+y}")
print(f"{x} menos {y} es {x-y}")
print(f"{x} por {y} es {x*y}")
print(f"{x} dividido {y} es {x/y}")
print(f"{z} dividido (truncando) por {y} es igual a {z//y}")
print(f"{z} módulo {y} es igual a {z%y}")
print(f"{x} elevado a {y} es {x**y}")
print(f"{x} elevado a {3} es {x**3}")
print(f"La raíz cuadrada de {x} es {x**0.5}")

6 más 2 es 8
6 menos 2 es 4
6 por 2 es 12
6 dividido 2 es 3.0
7 dividido (truncando) por 2 es igual a 3
7 módulo 2 es igual a 1
6 elevado a 2 es 36
6 elevado a 3 es 216
La raíz cuadrada de 6 es 2.449489742783178


## Integer y Floats

In [4]:
mi_numero = 1 + 3
print(mi_numero + mi_numero)
print(type(mi_numero))
mi_numero = 5.8
print(mi_numero)
print(type(mi_numero))

edad = input("Dime tu edad: ")
print("Tu edad es "+edad)

nueva_edad = 1 + edad  # Fallará porque edad es de tipo str
print("Vas a cumplir "+nueva_edad)

8
<class 'int'>
5.8
<class 'float'>
Tu edad es 24


TypeError: unsupported operand type(s) for +: 'int' and 'str'

## Conversiones

In [8]:
# implícitas

num1 = 20
num2 = 30.5

print(type(num1), type(num2))

num1 = num1 + num2

print(type(num1), type(num2))

# explícitas
num1 = 5.8
print(num1, type(num1))

num2 = int(num1)
print(num2, type(num2))

edad = input("Dime tu edad: ")

print(type(edad))

edad = int(edad)

print(type(edad))

nueva_edad = 1 + edad
print(nueva_edad, type(nueva_edad))
print("Tu nueva edad va a ser "+str(nueva_edad))

<class 'int'> <class 'float'>
<class 'float'> <class 'float'>
5.8 <class 'float'>
5 <class 'int'>
<class 'str'>
<class 'int'>
35 <class 'int'>
Tu nueva edad va a ser 35


## Redondeo

In [9]:
print(round(90/7))

valor = 95.66666666666666666666666666

valor = round(valor, 3)
print(valor)
print(type(valor))

valor = round(valor)
print(valor)
print(type(valor))

13
95.667
<class 'float'>
96
<class 'int'>


## Ejercicio propuesto
La situación es esta: Se acerca el Black Friday y en una tienda de tecnología se van a aplicar algunos descuentos. 
- Si el producto es de telefonía móvil, se le aplicará un 15% de descuento.
- Si el producto es de informática, se le aplicará un 18% de descuento.
- Si el producto es de cualquier otra categoría, se le aplicará un 22% de descuento.

Debes realizar un programa que solicite a un trabajador de la tienda el nombre del producto, la categoría y el precio y le indique el precio final para el Black Friday.